src/controller/java/MdnsImpl.cpp

```diff
@@ -0,0 +1,154 @@
+
+CHIP_ERROR ChipMdnsInit(MdnsAsyncReturnCallback initCallback, MdnsAsyncReturnCallback errorCallback, void * context)
+{
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ChipMdnsPublishService(const MdnsService * service)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR ChipMdnsStopPublish()
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR ChipMdnsStopPublishService(const MdnsService * service)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR ChipMdnsBrowse(const char * type, MdnsServiceProtocol protocol, Inet::IPAddressType addressType,
+                          Inet::InterfaceId interface, MdnsBrowseCallback callback, void * context)
+{
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+}
+
+CHIP_ERROR ChipMdnsResolve(MdnsService * service, Inet::InterfaceId interface, MdnsResolveCallback callback, void * context)
+{
+    VerifyOrReturnError(service != nullptr && callback != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(sResolverObject != nullptr && sResolveMethod != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    std::string serviceType = service->mType;
+    serviceType += '.';
+    serviceType += (service->mProtocol == MdnsServiceProtocol::kMdnsProtocolUdp ? "_udp" : "_tcp");
+
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    UtfString jniInstanceName(env, service->mName);
+    UtfString jniServiceType(env, serviceType.c_str());
+
+    env->CallVoidMethod(sResolverObject, sResolveMethod, jniInstanceName.jniValue(), jniServiceType.jniValue(),
+                        reinterpret_cast<jlong>(callback), reinterpret_cast<jlong>(context));
+
+    if (env->ExceptionCheck())
+    {
+        ChipLogError(Discovery, "Java exception in ChipMdnsResolve");
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+        return CHIP_JNI_ERROR_EXCEPTION_THROWN;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+// Implementation of other methods required by the CHIP stack
+
+void GetMdnsTimeout(timeval & timeout) {}
+void HandleMdnsTimeout() {}
+
+// Implemention of Java-specific functions
+
+void InitializeWithObject(jobject resolverObject)
+{
+    JNIEnv * env         = JniReferences::GetInstance().GetEnvForCurrentThread();
+    sResolverObject      = env->NewGlobalRef(resolverObject);
+    jclass resolverClass = env->GetObjectClass(sResolverObject);
+
+    VerifyOrReturn(resolverClass != nullptr, ChipLogError(Discovery, "Failed to get Resolver Java class"));
+
+    sResolveMethod = env->GetMethodID(resolverClass, "resolve", "(Ljava/lang/String;Ljava/lang/String;JJ)V");
+
+    if (sResolveMethod == nullptr)
+    {
+        ChipLogError(Discovery, "Failed to access Resolver 'resolve' method");
+        env->ExceptionClear();
+    }
+}
+
+void HandleResolve(jstring instanceName, jstring serviceType, jstring address, jint port, jlong callback, jlong context)
+{
+    VerifyOrReturn(callback != 0, ChipLogError(Discovery, "HandleResolve called with callback equal to nullptr"));
```

Comment:
Can we use `callback != nullptr` and same for the other `VerifyOrReturn` in this method?

Reply:
callback is of type `long` so it can't be compared to `nullptr`.